### PR TITLE
Feature/add administrate

### PR DESCRIPTION
**Pull request by noahsettersten**
Here I've added Administrate, along with a custom RolesField around our user roles as well as the helper used in a handful of our apps to allow hiding resources from the sidebar and fixing sorting parameters.

This is behind an option that can be declined to skip Administrate setup.

Additionally, this pulls in the new `administrate-field-trix` gem that I created this morning to make a `Field::Trix` available for immediate use in Administrate dashboard.

Issues #9, #26, and #27 are completed by this PR.

### CODE REVIEW

File: lib/voyage/templates/concerns_analytics_track.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      user_id: user.uuid,
+      event: event_name,
+      properties: {
+        browser: (browser.name rescue 'unknown'),
+        browser_id: (browser.id rescue 'unknown'),
+        browser_version: (browser.version rescue 'unknown'),
+        platform: (browser.platform rescue 'unknown'),
+        roles: (user.roles.map(&:to_s).join(',') rescue ''),
+        rails_env: Rails.env.to_s,
+      }.merge(sanitized_options),
```
Comment (GeekOnCoffee) [inline lib/voyage/templates/concerns_analytics_track.rb:29]: @noahsettersten this was on my list to clean up in another project, the rescues just seem so ugly; probably not in scope here though.

File: lib/voyage/templates/admin_application_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+module Admin
+  class ApplicationController < Administrate::ApplicationController
+    include AnalyticsTrack
+
+    before_action :authenticate_user!,
+                  unless: -> { is_a?(HighVoltage::PagesController) }
+    before_action :require_admin!
+
+    skip_authorization_check
+    impersonates :user
```
Comment (GeekOnCoffee) [inline lib/voyage/templates/admin_application_controller.rb:16]: Given that we're baking in impersonates, could we include the views in the admin to do the impersonation?

File: lib/voyage/templates/concerns_analytics_track.rb (hunk 1 above)
Comment (noahsettersten) [reply lib/voyage/templates/concerns_analytics_track.rb:29]: I agree, these rescues are holdovers from early on in Voyage but they were moved into a separate concern in this PR. Definitely something to address, I'll create a separate issue to keep track of it: #35.

File: lib/voyage/templates/admin_application_controller.rb (hunk 2 above)
Comment (noahsettersten) [reply lib/voyage/templates/admin_application_controller.rb:16]: Good point. In backporting these controller changes, I missed the respective Administrate view customizations as well.

**Pull request merged**